api/pkg/service/resource/resource.go

```diff
@@ -17,98 +17,120 @@ package resource
+	return &resource.Resources{}, nil
+}
+
+// VersionsByID returns all versions of a resource given its resource id
+func (s *service) VersionsByID(ctx context.Context, p *resource.VersionsByIDPayload) (*resource.VersionsByIDResult, error) {
+	return &resource.VersionsByIDResult{Location: fmt.Sprintf("/v1/resource/%d/versions", p.ID)}, nil
+}
+
+// Find resource using name of catalog & name, kind and version of resource
+func (s *service) ByCatalogKindNameVersion(ctx context.Context, p *resource.ByCatalogKindNameVersionPayload) (res *resource.ByCatalogKindNameVersionResult, err error) {
+	return &resource.ByCatalogKindNameVersionResult{Location: fmt.Sprintf("/v1/resource/%s/%s/%s/%s", p.Catalog, p.Kind, p.Name, p.Version)}, nil
+}
+
+// Find resource using name of catalog & name, kind and version of resource
+func (s *service) ByCatalogKindNameVersionReadme(ctx context.Context, p *resource.ByCatalogKindNameVersionReadmePayload) (res *resource.ByCatalogKindNameVersionReadmeResult, err error) {
+	return &resource.ByCatalogKindNameVersionReadmeResult{Location: fmt.Sprintf("/v1/resource/%s/%s/%s/%s/readme", p.Catalog, p.Kind, p.Name, p.Version)}, nil
+}
+
+// Find resource yaml using name of catalog & name, kind and version of resource
+func (s *service) ByCatalogKindNameVersionYaml(ctx context.Context, p *resource.ByCatalogKindNameVersionYamlPayload) (res *resource.ByCatalogKindNameVersionYamlResult, err error) {
+	return &resource.ByCatalogKindNameVersionYamlResult{Location: fmt.Sprintf("/v1/resource/%s/%s/%s/%s/yaml", p.Catalog, p.Kind, p.Name, p.Version)}, nil
+}
+
+// Find a resource using its version's id
+func (s *service) ByVersionID(ctx context.Context, p *resource.ByVersionIDPayload) (*resource.ByVersionIDResult, error) {
+	return &resource.ByVersionIDResult{Location: fmt.Sprintf("/v1/resource/version/%d", p.VersionID)}, nil
+}
+
+// Find resources using name of catalog, resource name and kind of resource
+func (s *service) ByCatalogKindName(ctx context.Context, p *resource.ByCatalogKindNamePayload) (*resource.ByCatalogKindNameResult, error) {
+	if p.Pipelinesversion != nil {
+		return &resource.ByCatalogKindNameResult{Location: fmt.Sprintf("/v1/resource/%s/%s/%s?pipelinesversion=%s", p.Catalog, p.Kind, p.Name, *p.Pipelinesversion)}, nil
 	}
-	res.Platforms = platforms
+	return &resource.ByCatalogKindNameResult{Location: fmt.Sprintf("/v1/resource/%s/%s/%s", p.Catalog, p.Kind, p.Name)}, nil
+}
 
-	res.HubURLPath = fmt.Sprintf("%s/%s/%s/%s", r.Resource.Catalog.Name, r.Resource.Kind, r.Resource.Name, r.Version)
+// Find a resource using it's id
+func (s *service) ByID(ctx context.Context, p *resource.ByIDPayload) (*resource.ByIDResult, error) {
+	return &resource.ByIDResult{Location: fmt.Sprintf("/v1/resource/%d", p.ID)}, nil
+}
 
-	return res
+func queryBuilder(queryParam, key, value string) string {
+	if len(queryParam) == 0 {
+		return queryParam + fmt.Sprintf("%s=%s", key, value)
```

Comment:
we can use
```
v := url.Values{}
v.Set()
```
a bit cleaner

Reply:
(could be a follow up pr)

Reply:
yeah sure will address it in a follow up pr